App crashes if all App permissions are disabled, or in a fresh install, in ChatActivity

**Actual Behaviour**

App crashes on ChatActivity if all permissions are disabled from inside onPause(). It looks like its due to textToSpeech being uninitialized during the start of the app, at this point.
Using a Moto G4 Plus, with Android 7.0

**Expected Behaviour**

The app shouldn't crash.

**Steps to reproduce it**

Disable all permissions separately (Android 6.0 and up) through Settings, and open the app. The app crashes once the ChatActivity is opened.

**LogCat for the issue**

10-01 03:57:41.863 21071-21071/? I/art: Late-enabling -Xcheck:jni
10-01 03:57:42.013 21071-21071/org.fossasia.susi.ai W/art: Before Android 4.1, method android.graphics.PorterDuffColorFilter android.support.graphics.drawable.VectorDrawableCompat.updateTintFilter(android.graphics.PorterDuffColorFilter, android.content.res.ColorStateList, android.graphics.PorterDuff$Mode) would have incorrectly overridden the package-private method in android.graphics.drawable.Drawable
10-01 03:57:42.206 21071-21071/org.fossasia.susi.ai I/TextInputLayout: EditText added is not a TextInputEditText. Please switch to using that class instead.
10-01 03:57:42.371 21071-21071/org.fossasia.susi.ai D/NetworkSecurityConfig: No Network Security Config specified, using platform default
10-01 03:57:42.536 21071-21121/org.fossasia.susi.ai I/Adreno: QUALCOMM build                   : 7d18700, I8ee426a9a2
                                                              Build Date                       : 10/07/16
                                                              OpenGL ES Shader Compiler Version: XE031.09.00.03
                                                              Local Branch                     : mybranch22308589
                                                              Remote Branch                    : quic/LA.BR.1.3.6_rb1.6
                                                              Remote Branch                    : NONE
                                                              Reconstruct Branch               : NOTHING
10-01 03:57:42.546 21071-21121/org.fossasia.susi.ai I/OpenGLRenderer: Initialized EGL, version 1.4
10-01 03:57:42.546 21071-21121/org.fossasia.susi.ai D/OpenGLRenderer: Swap behavior 1
10-01 03:57:42.571 21071-21071/org.fossasia.susi.ai D/AndroidRuntime: Shutting down VM
10-01 03:57:42.572 21071-21071/org.fossasia.susi.ai E/AndroidRuntime: FATAL EXCEPTION: main
                                                                      Process: org.fossasia.susi.ai, PID: 21071
                                                                      java.lang.RuntimeException: Unable to pause activity {org.fossasia.susi.ai/org.fossasia.susi.ai.chat.ChatActivity}: kotlin.UninitializedPropertyAccessException: lateinit property textToSpeech has not been initialized
                                                                          at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:3746)
                                                                          at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3712)
                                                                          at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3686)
                                                                          at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:3660)
                                                                          at android.app.ActivityThread.-wrap16(ActivityThread.java)
                                                                          at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1485)
                                                                          at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                          at android.os.Looper.loop(Looper.java:154)
                                                                          at android.app.ActivityThread.main(ActivityThread.java:6123)
                                                                          at java.lang.reflect.Method.invoke(Native Method)
                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
                                                                       Caused by: kotlin.UninitializedPropertyAccessException: lateinit property textToSpeech has not been initialized
                                                                          at org.fossasia.susi.ai.chat.ChatActivity.onPause(ChatActivity.kt:457)
                                                                          at android.app.Activity.performPause(Activity.java:6812)
                                                                          at android.app.Instrumentation.callActivityOnPause(Instrumentation.java:1344)
                                                                          at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:3735)
                                                                          at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3712) 
                                                                          at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3686) 
                                                                          at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:3660) 
                                                                          at android.app.ActivityThread.-wrap16(ActivityThread.java) 
                                                                          at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1485) 
                                                                          at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                          at android.os.Looper.loop(Looper.java:154) 
                                                                          at android.app.ActivityThread.main(ActivityThread.java:6123) 
                                                                          at java.lang.reflect.Method.invoke(Native Method) 
                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867) 
                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757) 

**Screenshots of the issue**
<img src="https://user-images.githubusercontent.com/11333048/31050186-d77ae122-a660-11e7-8920-261da57eed28.png" width="250">

**Would you like to work on the issue?**

Yes. Can someone confirm this issue is present?

--- a/app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt
+++ b/app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt
@@ -55,7 +55,7 @@ class ChatActivity: AppCompatActivity(), IChatView {
     lateinit var chatPresenter: IChatPresenter
     val PERM_REQ_CODE = 1
     lateinit var recyclerAdapter: ChatFeedRecyclerAdapter
-    lateinit var textToSpeech: TextToSpeech
+    var textToSpeech: TextToSpeech? = null
     var recordingThread: RecordingThread? = null
     lateinit var networkStateReceiver: BroadcastReceiver
     lateinit var progressDialog: ProgressDialog
@@ -63,7 +63,7 @@ class ChatActivity: AppCompatActivity(), IChatView {
 
     val afChangeListener = AudioManager.OnAudioFocusChangeListener { focusChange ->
         if (focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT || focusChange == AudioManager.AUDIOFOCUS_LOSS) {
-            textToSpeech.stop()
+            textToSpeech?.stop()
         }
     }
 
@@ -136,9 +136,7 @@ class ChatActivity: AppCompatActivity(), IChatView {
                 } else {
                     btnSpeak.setImageResource(R.drawable.ic_mic_24dp)
                     btnSpeak.setOnClickListener {
-                        if(textToSpeech.isSpeaking){
-                            textToSpeech.stop()
-                        }
+                        textToSpeech?.stop()
                         chatPresenter.startSpeechInput()
                     }
                 }
@@ -208,8 +206,8 @@ class ChatActivity: AppCompatActivity(), IChatView {
         Handler().post {
             textToSpeech = TextToSpeech(applicationContext, TextToSpeech.OnInitListener { status ->
                 if (status != TextToSpeech.ERROR) {
-                    val locale = textToSpeech.getLanguage()
-                    textToSpeech.language = locale
+                    val locale = textToSpeech?.getLanguage()
+                    textToSpeech?.language = locale
                 }
             })
         }
@@ -234,7 +232,7 @@ class ChatActivity: AppCompatActivity(), IChatView {
         handler.post {
             val result = audioFocus.requestAudioFocus(afChangeListener, AudioManager.STREAM_MUSIC, AudioManager.AUDIOFOCUS_GAIN)
             if (result == AudioManager.AUDIOFOCUS_REQUEST_GRANTED) {
-                textToSpeech.setOnUtteranceProgressListener(object : UtteranceProgressListener() {
+                textToSpeech?.setOnUtteranceProgressListener(object : UtteranceProgressListener() {
                     override fun onStart(s: String) {
                         if (recordingThread != null)
                             chatPresenter.stopHotwordDetection()
@@ -254,8 +252,8 @@ class ChatActivity: AppCompatActivity(), IChatView {
                 val ttsParams = HashMap<String, String>()
                 ttsParams.put(TextToSpeech.Engine.KEY_PARAM_UTTERANCE_ID,
                         this@ChatActivity.packageName)
-                textToSpeech.language = Locale(language)
-                textToSpeech.speak(reply, TextToSpeech.QUEUE_FLUSH, ttsParams)
+                textToSpeech?.language = Locale(language)
+                textToSpeech?.speak(reply, TextToSpeech.QUEUE_FLUSH, ttsParams)
                 audioFocus.abandonAudioFocus(afChangeListener)
             }
         }
@@ -316,9 +314,7 @@ class ChatActivity: AppCompatActivity(), IChatView {
             chatPresenter.check(true)
             btnSpeak.setImageResource(R.drawable.ic_mic_24dp)
             btnSpeak.setOnClickListener({
-                if(textToSpeech.isSpeaking){
-                    textToSpeech.stop()
-                }
+                textToSpeech?.stop()
                 chatPresenter.startSpeechInput()
             })
         } else {
@@ -454,7 +450,7 @@ class ChatActivity: AppCompatActivity(), IChatView {
         if (recordingThread != null)
             chatPresenter.stopHotwordDetection()
 
-        textToSpeech.stop()
+        textToSpeech?.stop()
 
     }
 
@@ -462,8 +458,8 @@ class ChatActivity: AppCompatActivity(), IChatView {
         super.onDestroy()
         rv_chat_feed.clearOnScrollListeners()
 
-        textToSpeech.setOnUtteranceProgressListener(null)
-        textToSpeech.shutdown()
+        textToSpeech?.setOnUtteranceProgressListener(null)
+        textToSpeech?.shutdown()
         chatPresenter.onDetach()
     }
 }

